open-craft/cloudera-zoho-reports
# [BB-2909] Replace OIDC with OAUTH2

This replaces OIDC with OAUTH2, because OIDC support has been removed as of version `3.0` of `auth-backends`.
This also upgrades Python to `3.8`, Django to `3.0.*` and all unpinned dependencies to their latest versions.

**Testing instructions**:
1. Start either Juniper or master devstack.
1. Navigate to http://localhost:18000/admin/oauth2_provider/application/add/ to add a new
   OAuth2 client.  Use these settings:
   - Client id: zoho_reports
   - Redirect uris: http://localhost:9000/complete/edx-oauth2/
   - Client type: Confidential
   - Authorization grant type: Authorization code
   - Client secret: open_secret
   - Name: zoho_reports
1. Navigate to http://localhost:18000/admin/oauth_dispatch/applicationaccess/add/ to add access for the new Oauth2 client.  Use these settings:
   - Application: zoho_reports
   - Scopes: user_id,profile,email
1. Make sure you have `pipenv` installed (or install it with e.g. `pip3 install --user pipenv`).
1. Set up virtualenv with `pipenv install`.
1. Run migrations (this will just create SQLite3 DB, so don't worry about the setup), create superuser (with other data than `edx` account from devstack) and start the zoho_reports development server using:
   ```bash
   pipenv run python manage.py migrate
   pipenv run python manage.py createsuperuser
   pipenv run python manage.py runserver 9000
   ```
1. Go to http://localhost:9000/admin/core/page/add/ and add a page with the following data:
   - URL path: edx
   - Page source: **see Jira comment for this**
   - Allowed users: edx@example.com
1. Log out from Django admin.
1. Navigate to http://localhost:9000/.
1. Log in with `edx@example.com` account and confirm authorization.
1. Check that the report is loading.

**Author notes and concerns**:
- [x] Add env variables to prod.
- [x] Set Application access on prod.
- [x] Remove `oidc` redirect URI from prod.

**Reviewers**
